### PR TITLE
2416 fix gitignore not generated bug

### LINKED ISSUE #2416: .gitignore not generated when init
### Please confirm that you have searched existing issues in the repo

Yes, I have searched the existing issues

### Any related issues?

_No response_

### Tell us about your environment

macOS 14

### MarkBind version

5.2.0

### Describe the bug and the steps to reproduce it

When running `npx markbind-cli init`, the `.gitignore` file is not generated. I tested on Windows 10 and it seems to work fine. May need someone to confirm using a mac machine.

### Expected behavior

Should generate .gitignore as per https://markbind-master.netlify.app/userguide/gitignorefile

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by LamJiuFong**
**What is the purpose of this pull request?**
Fixes #2416 

- [ ] Documentation update
- [x] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

**Overview of changes:**
Rename `.gitignore` to `gitignore` in template/default and rename `gitignore` to `.gitignore` when user run `markbind-cli init`

**Anything you'd like to highlight/discuss:**
I have tested it locally, but I think we can only know if this method actually works until we publish the package on npm, not sure if we are allowed to publish a draft package? 

**Testing instructions:**
Make the changes in this PR and `npm link` an empty folder to the local `markbind-cli`
Ensure that local markbind-cli is linked to local @markbind-core
eg, 
In the `packages/cli` directory, 
Run `npm ls --link` , should see `@markbind/core@5.3.0 -> ./packages/core`

In the empty folder, run `npx markbind-cli init`, a `.gitignore` file will be created

**Proposed commit message: (wrap lines at 72 characters)**
Change .gitignore of the default template to gitignore

The default template has a .gitignore file

.gitignore will be ignored by npm and is not able to be published.
The file will not be created in the users' project.

Let's rename it to gitignore so that it can be published and rename 
it back to .gitignore in the users' project.

---

**Checklist:** :ballot_box_with_check:

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No unrelated changes 

---

**Reviewer checklist:**

Indicate the [SEMVER](https://semver.org/) impact of the PR:
- [ ] Major (when you make incompatible API changes)
- [ ] Minor (when you add functionality in a backward compatible manner)
- [x] Patch (when you make backward compatible bug fixes)

At the end of the review, please label the PR with the appropriate label: `r.Major`, `r.Minor`, `r.Patch`.

Breaking change release note preparation (if applicable):
- To be included in the release note for any feature that is made obsolete/breaking

> Give a brief explanation note about:
>   - what was the old feature that was made obsolete
>   - any replacement feature (if any), and
>   - how the author should modify his website to migrate from the old feature to the replacement feature (if possible).

### CODE REVIEW

File: packages/core/template/default/gitignore [hunk 1]
```diff
@@ -20,4 +20,4 @@ _site/
 # IDE configs
 .vscode/
 .idea/*
-*.iml
\ No newline at end of file
+*.iml
```
Comment (yucheng11122017) [inline packages/core/template/default/gitignore:23]: unnecessary change

**Review by yucheng11122017 — COMMENTED** [review]
Thanks for the work @LamJiuFong and for the investigation as well! 

Generally looks good except for some minor nits.

**Comment by kaixin-hc** [comment]
formatting still got issue ah!

that aside, great investigation and thanks for finding a fix so quickly : )

File: packages/core/template/default/gitignore (hunk 1 above)
Comment (LamJiuFong) [reply packages/core/template/default/gitignore:23]: I am not sure how to fix this problem because the changes were made by WebStorm (due to missing EoF line in the original code), I tried using GitHub to remove the last line but nothing happened https://github.com/MarkBind/markbind/pull/2469/commits/d62c7ac70ba76651108d74e85f50028ef83c10dc

**Comment by LamJiuFong** [comment]
Thanks @kaixin-hc @yucheng11122017 for reviewing!

**Review by yucheng11122017 — APPROVED** [review]
LGTM! thanks!

**Review by kaixin-hc — APPROVED** [review]
lgtm ! thank u

**Comment by kaixin-hc** [comment]
@LamJiuFong Can you write a draft PR message that includes the explanation for why this fixes a bug? Thanks!

**Comment by LamJiuFong** [comment]
@kaixin-hc  I have updated the commit message, is that okay?
